pages/_app.js

```diff
@@ -1,20 +1,22 @@
 import Header from "@/components/Header";
 import GlobalStyle from "../styles";
-import useSWR from "swr";
+import useSWR, { SWRConfig } from "swr";
 
 const fetcher = (...args) => fetch(...args).then((res) => res.json());
 
 export default function App({ Component, pageProps }) {
-  const { data: snippets, error, isLoading } = useSWR("api/snippets", fetcher);
+  const { data, error, isLoading } = useSWR("api/snippets", fetcher);
```

Comment:
you already mentioned this in your recent GitHub Discussion:

You always fetch all snippets, regardless of the page.

It would make sense to have this fetch in your SnippetCardList component.

This would lead to only fetching all Snippets whenever the SnippetCardList component is called.
And to only fetching a single Snippet when in the Detail View